x-pack/plugins/code/server/code_node_client.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Server } from 'hapi';
+import { EsClient } from './lib/esqueue';
+
+export const CODE_NODE_INDEX_NAME = '.code_node';
```

Comment:
This index name won't allow the `kibana_system` role to interact with it: https://github.com/elastic/elasticsearch/blob/3ec0cc5f8ed0567593bf1bc2352ff164c2de662e/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java#L121-L122

It should be noted that if we change this index name to match the `.code-*` pattern which is used throughout the reserved roles, that `code_admin` will be able to write to this index.